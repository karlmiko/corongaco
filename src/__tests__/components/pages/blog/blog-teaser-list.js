import React from 'react'
import renderer from 'react-test-renderer'
import BlogTeaserList from '~components/pages/blog/blog-teaser-list'

const items = [
  {
    node: {
      authors: [
        {
          name: 'test name',
          twitterLink: 'https://twitter.com/test',
          headshot: {
            file: {
              fileName: 'file.png',
            },
            resize: {
              src: '//images.ctfassets.net/arandomfile',
            },
          },
        },
        {
          name: 'test name II',
          twitterLink: 'https://twitter.com/testii',
          headshot: {
            file: {
              fileName: 'file2.png',
            },
            resize: {
              src: '//images.ctfassets.net/aspecificfile',
            },
          },
        },
      ],
      publishDate: 'Aug 1 2020',
      slug: 'we-made-a-blog-post',
      title: 'We Made a Blog Post',
      lede: {
        lede: 'Huh, this blog thing seems to be working! This is the lede.',
      },
    },
  },
  {
    node: {
      authors: [
        {
          name: 'test name',
          twitterLink: 'https://twitter.com/test',
          headshot: {
            file: {
              fileName: 'file.png',
            },
            resize: {
              src: '//images.ctfassets.net/arandomfile',
            },
          },
        },
        {
          name: 'test name II',
          twitterLink: 'https://twitter.com/testii',
          headshot: {
            file: {
              fileName: 'file2.png',
            },
            resize: {
              src: '//images.ctfassets.net/aspecificfile',
            },
          },
        },
      ],
      publishDate: 'Apr 15 2020',
      slug: 'new-blog-post',
      title: 'New Blog Post',
      lede: {
        lede: 'This is a new blog post, not an old one.',
      },
    },
  },
]

describe('Components : Pages : Blog : Blog Teaser List', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<BlogTeaserList items={items} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
