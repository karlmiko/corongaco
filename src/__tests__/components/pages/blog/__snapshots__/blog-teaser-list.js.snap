// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components : Pages : Blog : Blog Teaser List renders correctly 1`] = `
Array [
  <h2
    className="hed-primary "
  >
    <a
      href="/blog/we-made-a-blog-post"
    >
      We Made a Blog Post
    </a>
  </h2>,
  <div
    className="bylineSmallMarginImage"
  >
    <div
      className="headshotContainer"
    >
      <img
        alt=""
        src="//images.ctfassets.net/arandomfile"
      />
      <img
        alt=""
        src="//images.ctfassets.net/aspecificfile"
      />
    </div>
    <p
      className="bylineText"
    >
      <span
        className="spacer"
      >
        By 
      </span>
      <span
        className="author"
      >
        <a
          className="authorInner"
          href="https://twitter.com/test"
        >
          test name
        </a>
         
        &
         
        <a
          className="authorInner"
          href="https://twitter.com/testii"
        >
          test name II
        </a>
      </span>
      <span
        className="date"
      >
        Aug 1 2020
      </span>
    </p>
  </div>,
  <p
    className="lede"
  >
    Huh, this blog thing seems to be working! This is the lede.
  </p>,
  <h2
    className="hed-primary "
  >
    <a
      href="/blog/new-blog-post"
    >
      New Blog Post
    </a>
  </h2>,
  <div
    className="bylineSmallMarginImage"
  >
    <div
      className="headshotContainer"
    >
      <img
        alt=""
        src="//images.ctfassets.net/arandomfile"
      />
      <img
        alt=""
        src="//images.ctfassets.net/aspecificfile"
      />
    </div>
    <p
      className="bylineText"
    >
      <span
        className="spacer"
      >
        By 
      </span>
      <span
        className="author"
      >
        <a
          className="authorInner"
          href="https://twitter.com/test"
        >
          test name
        </a>
         
        &
         
        <a
          className="authorInner"
          href="https://twitter.com/testii"
        >
          test name II
        </a>
      </span>
      <span
        className="date"
      >
        Apr 15 2020
      </span>
    </p>
  </div>,
  <p
    className="lede"
  >
    This is a new blog post, not an old one.
  </p>,
]
`;
